### PR TITLE
Master crm remove crm binding partner dbe tde

### PR DESCRIPTION
**Pull request by tde-banana-odoo**

### CODE REVIEW

File: addons/sale_crm/wizard/crm_opportunity_to_quotation.py [hunk 1]
```diff
@@ -19,13 +17,28 @@ def default_get(self, fields):
...
 
-        active_id = self._context.get('active_id')
-        if 'lead_id' in fields and active_id:
-            result['lead_id'] = active_id
+        lead = False
+        if result.get('lead_id'):
+            lead = self.env['crm.lead'].browse(result['lead_id'])
+        elif 'lead_id' in fields and self._context.get('active_id'):
+            lead = self.env['crm.lead'].browse(self._context['active_id'])
+        if lead:
```
Comment (dbeguin) [inline addons/sale_crm/wizard/crm_opportunity_to_quotation.py]:
```
lead = self.env['crm.lead'].browse(result.get('lead_id', 0))
if not lead and 'lead_id' in fields and self._context.get('active_id'):
    lead = self.env['crm.lead'].browse(self._context['active_id'])
if lead:
    ...
```
but maybe this is more indigest :thinking:

File: addons/crm/wizard/crm_lead_to_opportunity.py [hunk 2]
```diff
@@ -144,7 +153,7 @@ def _create_partner(self, lead_id, action, partner_id):
         #wizard and would probably diserve to be refactored or at least
         #moved to a better place
         if action == 'each_exist_or_create':
```
Comment (dbeguin) [inline addons/crm/wizard/crm_lead_to_opportunity.py]: This is used only in crm.lead2opportunity.partner.mass wizard
Maybe we could move this part to that wizard ?

File: addons/website_sms/models/website_visitor.py [hunk 3]
```diff
@@ -8,35 +8,35 @@
...
-                'number_field_name': 'mobile',
-            }
-        return {}
+    def _check_for_sms_composer(self):
+        """ Purpose of this method is to actualize visitor model prior to contacting
+        him. Used notably for inheritance purpose, when dealing with leads that
+        could update the visitor model. """
+        if not self.partner_id or (not self.partner_id.mobile and not self.partner_id.phone):
+            return False
+        return True
```
Comment (dbeguin) [inline addons/website_sms/models/website_visitor.py]: `return self.partner_id and (self.partner_id.mobile or self.partner_id.phone)`

File: addons/crm/models/crm_lead.py [hunk 4]
```diff
@@ -990,6 +990,34 @@ def _create_lead_partner(self):
...
+        if self.email_from:  # search through the existing partners based on the lead's email
+            return Partner.search([('email', '=', self.email_from)], limit=1).id
+
+        search_criteria = False
+        if self.partner_name:  # search through the existing partners based on the lead's partner or contact name
+            search_criteria = self.partner_name
+        elif self.contact_name:
+            search_criteria = self.contact_name
+        elif self.name:  # to be aligned with _create_lead_partner, search on lead's name as last possibility
+            search_criteria = self.name
```
Comment (dbeguin) [inline addons/crm/models/crm_lead.py]: `search_criteria = self.partner_name or self.contact_name or self.name`

File: addons/crm/wizard/crm_lead_to_opportunity.py [hunk 5]
```diff
@@ -41,20 +42,28 @@ def default_get(self, fields):
...
         return result
 
     name = fields.Selection([
         ('convert', 'Convert to opportunity'),
         ('merge', 'Merge with existing opportunities')
     ], 'Conversion Action', required=True)
+    action = fields.Selection([
+        ('create', 'Create a new customer'),
+        ('exist', 'Link to an existing customer'),
+        ('nothing', 'Do not link to a customer')
```
Comment (dbeguin) [inline addons/crm/wizard/crm_lead_to_opportunity.py]: This is added in partner.mass wizard. We should remove the addition in that wizard

Comment (tde-banana-odoo) [reply addons/crm/wizard/crm_lead_to_opportunity.py]: Done

File: addons/crm/wizard/crm_lead_to_opportunity.py (hunk 2 above)
Comment (tde-banana-odoo) [reply addons/crm/wizard/crm_lead_to_opportunity.py]: I have a task for that ! To be done in future commits.

File: addons/sale_crm/wizard/crm_opportunity_to_quotation.py (hunk 1 above)
Comment (tde-banana-odoo) [reply addons/sale_crm/wizard/crm_opportunity_to_quotation.py]: Mmh, browsing 0 is risky. I agree this is a bit heavy. Meh.

File: addons/website_sms/models/website_visitor.py (hunk 3 above)
Comment (tde-banana-odoo) [reply addons/website_sms/models/website_visitor.py]: Good idea. I updated code abotu mail composer at the same time.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

File: addons/crm/models/crm_lead.py (hunk 4 above)
Comment (dbeguin) [reply addons/crm/models/crm_lead.py]: :broken_heart:

**Pull request closed without merging**